### PR TITLE
osd, librados: add unset-manifest op

### PR DESCRIPTION
**Pull request by myoungwon**
This PR introduces a new rados op (unset-manifest).
Please refer to https://github.com/ceph/ceph/pull/19935.

 Signed-off-by: Myoungwon Oh <omwmw@sk.com>

### CODE REVIEW

**Comment by myoungwon** [comment]
@liewegas We need to discuss the remaining works and plan for dedup tier.
In my opinion, there are three main works.

1. chunk and hash implementation
- abstraction class and implementation for hash such as  SHA256 .... 
2. making a deduped object during writeback operations 
 - making fingerprints and inc/dec references
3. handling "leaking" reference
 - scrub process (https://pad.ceph.com/p/deduplication_how_do_we_store_chunk)

So, I would like to start remaining works in the above order.
What do you think?

**Comment by liewegas** [comment]
This plan sounds right!  Let's sketch out the abstraction/interfaces for the hashing and chunking..

File: src/test/librados/tier.cc [hunk 1]
```diff
@@ -3259,6 +3259,138 @@ TEST_F(LibRadosTwoPoolsPP, ManifestRefRead) {
...
+    ASSERT_EQ(0, completion->get_return_value());
+    completion->release();
+  }
+  // redirect's refcount 
+  {
+    bufferlist in, out;
+    cache_ioctx.exec("bar", "refcount", "chunk_read", in, out);
+    cls_chunk_refcount_read_ret read_ret;
+    try {
+      bufferlist::iterator iter = out.begin();
```
Comment (tchaikov) [inline src/test/librados/tier.cc]: because of the recent `bufferlist::const_iterator` change, please use `out.cbegin()`

**Comment by tchaikov** [comment]
retest this please.

File: src/test/librados/tier.cc (hunk 1 above)
Comment (tchaikov) [reply src/test/librados/tier.cc]: see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/12075//consoleFull

**Comment by myoungwon** [comment]
@tchaikov Fixed. Can you take a look?

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-06-05_05:07:24-rados-wip-kefu-testing-2018-06-05-0923-distro-basic-smithi/

**Pull request merged**